src/components/map/RealtimeLayersService.js

```diff
@@ -51,42 +50,41 @@ goog.require('ga_map_service');
 
         scope.$watchCollection('layers | filter:layerFilter',
             function(newLayers, oldLayers) {
+
           // Layer Removed
-          for (var i = 0; i < oldLayers.length; i++) {
-            var bodId = oldLayers[i].bodId;
+          oldLayers.forEach(function(oldLayer) {
+            var bodId = oldLayer.bodId;
             var oldLayerIdIndex = realTimeLayersId.indexOf(bodId);
             if (oldLayerIdIndex != -1 &&
                 !gaMapUtils.getMapLayerForBodId(map, bodId)) {
               realTimeLayersId.splice(oldLayerIdIndex, 1);
-              $timeout.cancel(timers.splice(oldLayerIdIndex, 1));
+              $timeout.cancel(timers.splice(oldLayerIdIndex, 1)[0]);
```

Comment:
A minor one if you removed an updatable  Geojson layer, the next request for the update was triggered after the delay. With this, the useless request is correctly canceled.

The splice method returns an array and $timeout.cancel takes a promise in parameter not an array of promise.